indico/modules/events/registration/clone.py

```diff
@@ -22,47 +22,114 @@
 from indico.modules.events.registration.models.forms import RegistrationForm
 from indico.modules.events.registration.models.form_fields import RegistrationFormFieldData
 from indico.modules.events.registration.models.items import RegistrationFormItem, RegistrationFormSection
+from indico.modules.events.registration.models.registrations import Registration, RegistrationData
 from indico.util.i18n import _
 
 from MaKaC.conference import EventCloner
 
 
+def _find_registration_forms(event):
+        return RegistrationForm.find(~RegistrationForm.is_deleted,
+                                     RegistrationForm.event_id == int(event.id))
+
+
 class RegistrationFormCloner(EventCloner):
     def get_options(self):
         if not is_feature_enabled(self.event, 'registration'):
             return {}
-        enabled = bool(self._find_registration_forms().count())
+        enabled = bool(_find_registration_forms(self.event).count())
         return {'form': (_('Registration forms'), enabled, False)}
 
     def clone(self, new_event, options):
         if 'form' not in options:
             return
-        attrs = get_simple_column_attrs(RegistrationForm) - {'start_dt', 'end_dt'}
-        for old_form in self._find_registration_forms():
+        attrs = get_simple_column_attrs(RegistrationForm) - {'event_id', 'start_dt', 'end_dt'}
```

Comment:
`event_id` should be automatically excluded since `get_simple_column_attrs` skips both the PK and FKs